extension/filetemplateextension/factory.go

```diff
@@ -0,0 +1,53 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package filetemplateextension // import "github.com/elastic/opentelemetry-collector-components/extension/filetemplateextension"
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/extension"
+)
+
+var (
+	extensionType = component.MustNewType("file_templates")
+)
+
+const (
+	extensionStability = component.StabilityLevelAlpha
```

Comment:
Should we mark all the components as "development" stability? I think this is preferred when they haven't been included in a distribution yet or breaking changes are expected. 

```suggestion
	extensionStability = component.StabilityLevelDevelopment
```

In addition, it would be great if you could add the "metadata.yaml" file in the component to dynamically generate all this metadata.
Example file: https://github.com/elastic/opentelemetry-collector-components/blob/main/processor/elasticinframetricsprocessor/metadata.yaml
File to run go generate: https://github.com/elastic/opentelemetry-collector-components/blob/main/processor/elasticinframetricsprocessor/doc.go#L18

Let's sync if you need some guidance on this :)

Reply:
Added the metadata files, but not sure how to use them to generate anything from it :slightly_smiling_face: Are there docs about this?

Reply:
Looking for the docs, but these should be the steps:

- Add the `doc.go` file in the component (e.g. https://github.com/elastic/opentelemetry-collector-components/blob/main/processor/elasticinframetricsprocessor/doc.go#L18) -> the important field here is the "go generate" statement insdie the file which will call the mdatagen tool
- Run "make generate" and the "./internal" files should be added

Reply:
https://github.com/open-telemetry/opentelemetry-collector/blob/main/cmd/mdatagen/README.md

Reply:
Thanks! Code updated to use this.